src/Bicep.Core/Semantics/Metadata/DeclaredResourceMetadata.cs

```diff
@@ -14,7 +14,8 @@ public record DeclaredResourceMetadata(
         ResourceType Type,
         bool IsExistingResource,
         ResourceSymbol Symbol,
-        ResourceMetadataParent? Parent)
+        ResourceMetadataParent? Parent,
+        TernaryOperationSyntax? TernaryOperationSyntax)
```

Comment:
It's unclear from the property names here that `TernaryOperationSyntax` applies to the resource parent (rather than in some other way to the declared resource). The role of this property could be made clearer by changing its name or by adding it to the `ResourceMetadataParent` record (which represents the `parent` expression) rather than the `DeclaredResourceMetadata` (which represents the full resource).

Reply:
Good point!